quay_setup.sh

```diff
@@ -1,18 +1,56 @@
 # oc create -f https://raw.githubusercontent.com/minio/minio-operator/master/minio-operator.yaml
+
+check(){
+  eval VAR1="$1"
+  eval VAR2="$2"
+  while true; do
+    eval $VAR1
+    if [[ $? == $VAR2 ]]
+    then
+      break
+    fi
+    sleep 1
+  done
+}
+
+#Create a Project
+oc new-project quay-enterprise
 # install quay operator
 oc apply -f quay-op.yaml
-# install cso operator
-oc apply -f cso.yaml
 # add pull secrete for quay images
 oc create secret generic redhat-pull-secret --from-file=".dockerconfigjson=docker_quay.json" --type='kubernetes.io/dockerconfigjson'
+#Quay-operator takes some time to come(especially on crc) up so need to wait before bringing up quayecosystem
+
+#code to wait for quay-operator to come up
+req="1"
+com="oc get pods -o json | jq -r '.items[0].status.phase' | grep -v Running"
+check "\${com}" "\${req}"
+
 # create quay instance
 oc apply -f quayecosystem.yaml
-# enable clair v4 scanning for clairv4 org
-# this has to be done after creation of quay-enterprise-config-secret
-oc patch secret quay-enterprise-config-secret -p "$(oc get secrets quay-enterprise-config-secret -o json | jq -r '.data["config.yaml"] | @base64d | . + "SECURITY_SCANNER_V4_ENDPOINT: http://clairv4\nSECURITY_SCANNER_V4_NAMESPACE_WHITELIST:\n- clairv4\n\n"| @base64 | {"data":{"config.yaml": .}}')"
+
 # add clair v4 config
 oc create secret generic con-clair --from-file=./config.yaml
 # clair v4 supporting services
 oc apply -f jager.yaml
 oc apply -f clairv4_db.yaml
 oc apply -f clairv4.yaml
+
+
+#Check if secret is created
+req='0'
+com="oc get secrets quay-enterprise-config-secret"
+check "\${com}" "\${req}"
```

Comment:
```suggestion
exec_and_wait "oc get secrets quay-enterprise-config-secret"
```

Reply:
Also I dont think we can pass arguments like -> ```exec_and_wait "oc get secrets quay-enterprise-config-secret"``` It gets stored in different args parsed according to " " in $1,$2....